src/owner.rs

```diff
@@ -172,47 +215,48 @@ impl Team {
         let (handle, resp) = http::github(app, &url, &token)?;
         let org: Org = http::parse_github_response(handle, &resp)?;
 
-        Team::insert(conn, login, team.id, team.name, org.avatar_url)
+        NewTeam::new(login, team.id, team.name, org.avatar_url).create_or_update(&conn)
     }
 
-    pub fn insert(
-        conn: &PgConnection,
-        login: &str,
-        github_id: i32,
-        name: Option<String>,
-        avatar: Option<String>,
-    ) -> CargoResult<Self> {
-        use diesel::pg::upsert::*;
+    /// Phones home to Github to ask if this User is a member of the given team.
+    /// Note that we're assuming that the given user is the one interested in
+    /// the answer. If this is not the case, then we could accidentally leak
+    /// private membership information here.
+    pub fn contains_user(&self, app: &App, user: &User) -> CargoResult<bool> {
+        team_with_gh_id_contains_user(app, self.github_id, user)
+    }
 
-        #[derive(Insertable, AsChangeset)]
-        #[table_name = "teams"]
-        struct NewTeam<'a> {
-            login: &'a str,
-            github_id: i32,
-            name: Option<String>,
-            avatar: Option<String>,
-        }
-        let new_team = NewTeam {
+    pub fn encodable(self) -> EncodableTeam {
+        let Team {
+            id,
+            name,
+            login,
+            avatar,
+            ..
+        } = self;
+        let org_name = Team::get_org(&login);
+        let team_name = Team::get_team(&login);
+        let url = format!("https://github.com/orgs/{}/teams/{}", org_name, team_name);
+
+        EncodableTeam {
+            id: id,
             login: login,
-            github_id: github_id,
             name: name,
             avatar: avatar,
-        };
+            url: Some(url),
+        }
+    }
 
-        diesel::insert(&new_team.on_conflict(
-            teams::github_id,
-            do_update().set(&new_team),
-        )).into(teams::table)
-            .get_result(conn)
-            .map_err(Into::into)
+    fn get_org(login: &str) -> String {
```

Comment:
I think we can refactor this a bit into one function that returns two string slices in a tuple-- I'll explain what I mean in person :)